src/time_dependent_operator.jl

```diff
@@ -0,0 +1,302 @@
+
+import Base: size, *, +, -, /, ==, isequal, adjoint, convert
+
+abstract type AbstractTimeDependentOperator{BL,BR} <: AbstractOperator{BL,BR} end
+
+current_time(::AbstractOperator) = throw(ArgumentError("Time not defined for operator."))
```

Comment:
Recently I have started taking cues from the SciML ecosystem and trying to be very detailed in my error messages, suggesting possible workarounds and giving more explanations. This has a very high value in how novices perceive a library. Could you consider expanding this to something like "Time is not defined for this operator. You are probably attempting to use a time-independent operator in a code pathway that expects explicitly time-dependent operators. Consider wrapping ... ...".